### PR TITLE
Editorial: Improve consistency of property and record field names

### PR DESCRIPTION
**Pull request by gibson042**
* `"…" property` → `*"…"* property`
* ``` `…` property``` → `*"…"* property`
* `[[…]] property` → `[[…]] field`

### CODE REVIEW

**Review by michaelficarra — APPROVED** [review]
Not totally sure about the `SuperProperty` change, but the rest LGTM.

File: spec.html [hunk 1]
```diff
@@ -10669,7 +10669,7 @@ <h1>Function Environment Records</h1>
                 GetSuperBase()
               </td>
               <td>
-                Return the object that is the base for `super` property accesses bound in this Environment Record. The value *undefined* indicates that `super` property accesses will produce runtime errors.
+                Return the object that is the base for |SuperProperty| accesses bound in this Environment Record. The value *undefined* indicates that such accesses will produce runtime errors.
```
Comment (bakkot) [inline spec.html:10672]: I'm fine with the `such` change, but I do prefer the original `` `super` property`` phrasing.

File: spec.html [hunk 2]
```diff
@@ -13177,7 +13177,7 @@ <h1>ECMAScript Function Objects</h1>
             an Object
           </td>
           <td>
-            If the function uses `super`, this is the object whose [[GetPrototypeOf]] provides the object where `super` property lookups begin.
+            If the function uses `super`, this is the object whose [[GetPrototypeOf]] provides the object where |SuperProperty| lookups begin.
```
Comment (bakkot) [inline spec.html:13180]: Ditto above.

**Review by syg — APPROVED** [review]
lgtm % @bakkot and @michaelficarra's comment about super.

**Review by bakkot — APPROVED** [review]
LGTM other than the `super` things.

File: spec.html (hunk 1 above)
Comment (gibson042) [reply spec.html]: Restored, but it still reads a little awkwardly to me. What would @tc39/ecma262-editors think about adding parenthetical clarification?
```suggestion
                Return the object that is the base for `super` property accesses bound in this Environment Record (i.e., instances of |SuperProperty|). The value *undefined* indicates that evaluation of such accesses will produce runtime errors.
```

Comment (bakkot) [reply spec.html:10672]: The parenthetical doesn't really seem helpful to me personally.

**Pull request merged**